java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -2845,21 +3122,181 @@ protected boolean cleanImpl(boolean logErrorIfNotClean) {
 
     @Override
     public boolean isClean() {
-      return viewHandle == 0 && columnHandle == 0 && data == null && valid == null && offsets == null;
+      return viewHandle == 0 && columnHandle == 0 && toClose.isEmpty();
     }
 
     /**
      * This returns total memory allocated in device for the ColumnVector.
      * @return number of device bytes allocated for this column
      */
     public long getDeviceMemorySize() {
+      BaseDeviceMemoryBuffer valid = getValid();
+      BaseDeviceMemoryBuffer data = getData();
+      BaseDeviceMemoryBuffer offsets = getOffsets();
       long size = valid != null ? valid.getLength() : 0;
       size += offsets != null ? offsets.getLength() : 0;
       size += data != null ? data.getLength() : 0;
       return size;
     }
   }
 
+  public static ColumnVector createNestedColumnVector(DType type, int rows, HostMemoryBuffer data, HostMemoryBuffer valid, HostMemoryBuffer offsets,
+                                                      Optional<Long> nullCount, HostColumnVector.NestedHostColumnVector child) {
+    return NestedColumnVector.createColumnVector(type, rows, data, valid, offsets, nullCount, child);
+  }
+
+  private static class NestedColumnVector {
+
+    private final DeviceMemoryBuffer data;
+    private final DeviceMemoryBuffer valid;
+    private final DeviceMemoryBuffer offsets;
+    private final DType dataType;
+    private final int rows;
+    private final Optional<Long> nullCount;
+    NestedColumnVector child;
+
+    private NestedColumnVector(DType type, int rows, Optional<Long> nullCount,
+                               DeviceMemoryBuffer data, DeviceMemoryBuffer valid,
+                               DeviceMemoryBuffer offsets, NestedColumnVector child) {
+      this.dataType = type;
+      this.rows = rows;
+      this.nullCount = nullCount;
+      this.data = data;
+      this.valid = valid;
+      this.offsets = offsets;
+      this.child = child;
+    }
+
+    static ColumnVector createColumnVector(DType type, int rows, HostMemoryBuffer data, HostMemoryBuffer valid, HostMemoryBuffer offsets,
```

Comment:
Can we get some java docs on these.  It is not really clear why this API is hear instead of a part of `ColumnVector` itself.  It is also not 100% clear what this API does.